firedrake/slate/slac/kernel_builder.py

```diff
@@ -125,20 +126,20 @@ def prepare_tsfc_kernels(temps, tsfc_parameters=None):
     for expr in temps.keys():
         integrals = expr.form.integrals()
         mapper = RemoveRestrictions()
-        integrals = map(partial(map_integrand_dags, mapper), integrals)
+        integrals = list(map(partial(map_integrand_dags, mapper), integrals))
         prefix = "subkernel%d_" % len(kernel_exprs)
 
         # Now we split integrals by type: interior_facet and all other cases
         # First, the interior_facet case:
-        interior_facet_intergrals = filter(lambda x: x.integral_type() == "interior_facet", integrals)
+        interior_facet_intergrals = list(filter(lambda x: x.integral_type() == "interior_facet", integrals))
```

Comment:
Sure, if it is only used once, it will work. However, if one adds another usage later, the failure mode is quite unpleasant: you get what you expect at the first iteration, and then you get no elements for any further iteration attempts.